### ISSUE TITLE
Cannot update on FreeBSD

### ISSUE THREAD

**Issue by alexreg** [rust-lang/rustup#1391]
Rustup has not been able to update Rust on FreeBSD for a while now. I get this error message:

```
$ rustup update
info: syncing channel updates for 'stable-x86_64-unknown-freebsd'
info: syncing channel updates for 'nightly-x86_64-unknown-freebsd'
info: latest update on 2018-04-08, rust version 1.27.0-nightly (056f589fb 2018-04-07)
error: component 'rust-docs' for 'x86_64-unknown-freebsd' is unavailable for download
info: checking for self-updates

       stable-x86_64-unknown-freebsd unchanged - rustc 1.25.0 (84203cac6 2018-03-25)
  nightly-x86_64-unknown-freebsd update failed - rustc 1.26.0-nightly (75af15ee6 2018-03-20)
```

So, the docs component is presumably missing (not built?), and this is causing the update to fail.

**Comment by thefallenidealist** [comment]
Try to remove ~/.rustup and ~/.cargo and run script from https://rustup.rs/

```
%  rustup show
Default host: x86_64-unknown-freebsd

nightly-x86_64-unknown-freebsd (default)
rustc 1.27.0-nightly (7360d6dd6 2018-04-15)
```

**Comment by alexreg** [comment]
Thanks, that does the trick. Not sure how things got corrupted!

**Comment by asomers** [comment]
I'm experiencing this issue on a fresh jail with an empty .rustup and .cargo dir.  I can install nightly, but not stable.
```
>rustup toolchain install stable
info: syncing channel updates for 'stable-x86_64-unknown-freebsd'
207.2 KiB / 260.2 KiB ( 80 %)   0 B/s ETA: Unknown                260.2 KiB / 260.2 KiB (100 %) 207.2 KiB/s ETA:   0 s                
info: latest update on 2018-05-10, rust version 1.26.0 (a77568041 2018-05-07)
error: component 'rust-docs' for 'x86_64-unknown-freebsd' is unavailable for download
```

**Comment by asomers** [comment]
This is a duplicate of https://github.com/rust-lang/rust/issues/49462 .

**Comment by mqudsi** [comment]
@asomers no, it's not, actually. Even the nightly toolchain doesn't install without deleting `~/.rustup` under freebsd.

**Comment by Diggsey** [comment]
Fixed by #1419
